northstar/src/api/client.rs

```diff
@@ -539,17 +502,16 @@ impl<'a, T: AsyncRead + AsyncWrite + Unpin> Client<T> {
     /// # #[tokio::main(flavor = "current_thread")]
     /// # async fn main() {
     /// let mut client = Client::new(tokio::net::TcpStream::connect("localhost:4200").await.unwrap(), None, Duration::from_secs(10)).await.unwrap();
-    /// client.mount(vec!("test:0.0.1")).await.expect("Failed to mount");
+    /// client.mount_all(vec!("hello-world:0.0.1", "cpueater:0.0.1")).await.expect("Failed to mount");
     /// # }
     /// ```
-    pub async fn mount<E, C, I>(&mut self, containers: I) -> Result<Vec<MountResult>, Error>
+    pub async fn mount_all<E, C, I>(&mut self, containers: I) -> Result<Vec<MountResult>, Error>
```

Comment:
Then you need "collected" collection. The reason of `I` is that I encountered often the case that I request the container list from the rt and iterate and filter. With the current signature I can directly pass the iter to to the client interface.

A subject for being changed is the bounds which are super confusing because of the `Error` type requirement that needs to match the `TryInto` impl for `Name`, `Container`, `NonNullString`.

Mabe replace all those with a simple `ToString` bound and take the penalty of the String allocation? Can we discuss that in a dedicated ticket?